### ISSUE TITLE
Negative dimension size caused by subtracting 7 from 3 for 'avg_pool/AvgPool' (op: 'AvgPool') with input shapes: [?,3,3,2048].

### ISSUE THREAD

**Issue by lmbak** [azavea/raster-vision#808]
This error occurs for the chip classification.  I'm guessing it has something to do with the input data? ([link](https://github.com/keras-team/keras/issues/3945))

edit: the solutions proposed via above link, both via editting the ~/.keras/keras.json or keras.backend.set_image_dim_ordering('th') did not work. Also, both the downloaded resnet50 seems to use the tf dim ordering because it is named "resnet50_weights_**tf_dim_ordering**_tf_kernels_notop.h5". So setting the dim ordering to 'th' does not make much sense.

I'm using spacenet input data with a 650x650 dimensions, with 3 bands. This seems to be the same as the input in the raster-vision examples. During the process the data gets automagically converted from 16 bit to 8 bit.

Also, the dimensions of [?,3,3,2048] do not seem to be in the resnet 50 model as specified by [fchollet](https://github.com/fchollet/deep-learning-models/blob/master/resnet50.py), which if i'm not mistaking is used as source for the resnet 50 model.

Is the problem indeed in the input data? If so, how can the problem be resolved? If not in the input data, then what is causing the problem?

**Comment by lmbak** [comment]
[Possibly related](https://github.com/fchollet/deep-learning-models/issues/76), especially since the error is identical and I'm also using a resnet50. Investigating further....

**Comment by lmbak** [comment]
For an image with dimensions of 650x650 the error is thrown if the chip size is set to 100x100. When set to 650x650 the problem does not occur. So it indeed seems to be the problem as referenced in the above post. For now the solution is to set your chip size not to be too small.

**Comment by lewfish** [comment]
Some networks have a minimum input size, which often isn't well documented. In this case, the Resnet50 divides the input size by 2 5 times because there are 5 max pooling layers. With an input size of 100, you're down to 3x3. And then there is an AvgPooling layer which operates over 7x7 patches, which incompatible with 3x3 input. I think more recent implementation using AdaptiveAvgPooling which gets around this issue by producing output of a fixed size regardless of the input size. I'll add something to the docs about this.
